### DISCUSSION TITLE
Customize evaluation time using cron-style evaluation intervals

### DISCUSSION THREAD

**Discussion by lex-pec** [prometheus/prometheus#12082] [Ideas]
Hello, 
I am an indirect user of Prometheus, specifically Grafana uses the same alert generator specification as Prometheus to evaluate alerts. So when I tried to create [a discussion](https://github.com/grafana/grafana/discussions/63090) on this topic on Grafana I was redirected to your project.

**Feature description:**
The idea is to add the ability to configure evaluation time of an alert not only periodically: `every X s/m/h`, but also via cron-style evaluation intervals like `at 17:30 every day` or `at 10:00 every Monday`.

**Why this feature:** 
I use alerts provided by Prometheus not only for monitoring the health of the system or its parts, but also as some kind of reports, so I can see what is going on at the specefic time every day and the only way to make it possible is to start the alert at this time, but it is not very convenient.

**Comment by WellyngtonF** [comment]
Can we get some attention on this idea ?

**Comment by machine424** [comment]
Have you considered incorporating the time restrictions as conditions in your expression? Check the second part of https://www.robustperception.io/combining-alert-conditions/.

Ensure that your alerts cannot be easily missed (If prometheus restarts etc.) and remain vigilant for any changes in time.

You can also obtain these reports by directly connecting to the Prometheus API at the specified times and evaluating your expressions. There may already be tools available that offer this functionality.
